convert2rhel/actions/system_checks/check_firewalld_availability.py

```diff
@@ -0,0 +1,119 @@
+    :rtype: bool
+    """
+    # Return false if the config file does not exist. Either it means that
+    # firewalld is not installed or the config file was removed.
+    if not os.path.exists(FIREWALLD_CONFIG_FILE):
+        logger.debug("%s does not exist." % FIREWALLD_CONFIG_FILE)
+        return False
+
+    contents = ""
+    with open(FIREWALLD_CONFIG_FILE, mode="r") as handler:
+        contents = [line.strip() for line in handler.readlines() if line.strip()]
+
+    # Contents list is empty for some reason, better to assume that there
+    # is no content in the file that was read.
+    if not contents:
+        logger.debug("%s is empty." % FIREWALLD_CONFIG_FILE)
+        return True
+
+    # If the config file has this option set to true/yes, then we need to
+    # return True to ask the user to change it to False.
+    if list(filter(CLEANUP_MODULES_ON_EXIT_REGEX.match, contents)):
+        logger.debug("CleanupModulesOnExit option enabled in %s" % FIREWALLD_CONFIG_FILE)
+        return True
+
+    logger.debug(
+        "Couldn't find CleanupModulesOnExit in firewalld.conf. Treating it as enabled because of default behavior."
+    )
+    # Return true anyway at the end even if the option was not found as the
+    # default value for CleanupModulesOnExit in the firewalld.conf is to be
+    # enabled always.
+    return True
+
+
+class CheckFirewalldAvailability(actions.Action):
+    id = "CHECK_FIREWALLD_AVAILABILITY"
+
+    def run(self):
+        """Error out if the firewalld service is running on the system."""
+        super(CheckFirewalldAvailability, self).run()
+        logger.task("Prepare: Check that firewalld is running")
+
+        if system_info.id != "oracle" and system_info.version.major != 8 and system_info.version.minor < 8:
+            logger.info("Skipping the check as it is relevant only for Oracle Linux 8.8 and above.")
+            return
+        if not systeminfo.is_systemd_managed_service_running("firewalld"):
+            logger.info("Firewalld service reported that it is not running.")
+            return
+        if _is_modules_cleanup_enabled():
+                    self.set_result(
+                        level="ERROR",
+                        id="FIREWALLD_MODULES_CLEANUP_ON_EXIT_CONFIG",
+                        title="Firewalld is set to cleanup modules after exit.",
+                        description="Firewalld running on Oracle Linux 8 can lead to a conversion failure.",
+                        diagnosis="We've detected that firewalld unit is running and that causes iptables and nftables failures on Oracle Linux 8 and under certain conditions it can lead to a conversion failure.",
+                        remediation=(
+                            "Set the option CleanupModulesOnExit in /etc/firewalld/firewalld.conf to no prior to running convert2rhel:\n"
+                            " sed -i -- 's/CleanupModulesOnExit=yes/CleanupModulesOnExit=no/g' /etc/firewalld/firewalld.conf\n"
```

Comment:
@r0x0d I remember you told me that even the `firewall-cmd --reload` command should be used. Shouldn't we mention this here as well?

Reply:
good catch!